admin_guide/sdn_troubleshooting.adoc

```diff
@@ -0,0 +1,527 @@
+* Are your duplex settings, and ethernet speeds right (on both ends)? (Look through the rest of the `ethtool _interface_` information)
+* Are the cables plugged in correctly?  To the correct ports?
+* Are the switches configured correctly?
+
+
+Once you have ascertained that the node to node connectivity is fine,
+we need to look at the SDN configuration on both ends.  Please proceed
+to the <<debugging-local-networking>> section.
+
+
+== Debugging Local Networking
+
+At this point we should have a list of one or more endpoints that you
+can't communicate with, but that have node to node connectivity.  For
+each one, we need to work out what is wrong, but first you need to
+understand how the SDN sets up the networking on a node for the
+different pods.
+
+=== The Interfaces on a Node
+
+These are the interfaces that the OpenShift SDN creates:
+
+* `br0`: The OVS bridge device that containers will be attached to.
+OpenShift SDN also configures a set of non-subnet-specific flow rules
+on this bridge. (The `multitenant` plugin does this immediately; the
+`ovssubnet` plugin waits until the SDN master announces the creation
+of the new node subnet.)
+* `lbr0`: A Linux bridge device, which is configured as Docker's bridge and
+given the cluster subnet gateway address (eg, 10.1.x.1/24).
+* `tun0`: An OVS internal port (port 2 on `br0`). This *also* gets assigned the
+cluster subnet gateway address, and is used for external network
+access. OpenShift SDN configures `netfilter` and routing rules to enable access
+from the cluster subnet to the external network via NAT.
+* `vlinuxbr` and `vovsbr`: Two Linux peer virtual Ethernet interfaces.
+`vlinuxbr` is added to `lbr0`, and `vovsbr` is added to `br0` (port 9), to
+provide connectivity for containers created directly with Docker outside of
+OpenShift.
+* `vxlan0`: The OVS VXLAN device (port 1 on `br0`), which provides access to
+containers on remote nodes.
+* `vethX` (in the main netns): A Linux virtual ethernet peer of `eth0` in the docker netns.  It will be attached to the OVS bridge on one of the ports 3-8 or 10.
+
+
+=== SDN Flows Inside a Node
+
+image:SDN.png[The possible SDN paths]
+
+Depending on what you are trying to access (or be accessed from) the
+path will vary.  There are four different places the SDN connects
+(inside a node).  They are labeled in red on the diagram above.
+
+* *Pod:* Traffic is going from one pod to another on the same machine (1 to a different 1)
+* *Remote Node (or Pod):* Traffic is going from a local pod to a remote node or pod in the same cluster (1 to 2)
+* *External Machine:* Traffic is going from a local pod outside the cluster (1 to 3)
+* *Local Docker:* Traffic is going from a local pod to a local docker container that is not managed by Kubernetes (1 to 4)
+
+=== Debugging Steps
+
+==== Is IP Forwarding Enabled?
+
+Check that `sysctl net.ipv4.ip_forward` is set to 1 (and check the host if this is a VM)
```

Comment:
It shouldn't really be possible for this to not be set; the setup script sets it, and if an error occurred before that point in the script, then openshift should fail to start